pkg/apis/garden/validation/validation.go

```diff
@@ -466,7 +466,7 @@ func ValidateSeedSpec(seedSpec *garden.SeedSpec, fldPath *field.Path) field.Erro
 		allErrs = append(allErrs, field.Required(cloudPath.Child("region"), "must provide a region"))
 	}
 
-	r, _ := regexp.Compile(`^(?:[a-zA-Z0-9\-]+\.)*[a-zA-Z0-9]+\.[a-zA-Z0-9]{2,6}$`)
+	r, _ := regexp.Compile(`^(?:[a-zA-Z0-9\-]+\.)*[a-zA-Z0-9\-]+\.[a-zA-Z0-9]{2,6}$`)
```

Comment:
This will match `foo-.com` which is not a valid domain.
`validation.go` has a utility function for validation of domains:

https://github.com/gardener/gardener/blob/37665ea119c94b2deb864e561dd68f6c016887e5/pkg/apis/garden/validation/validation.go#L1308-L1317

and usage: 

https://github.com/gardener/gardener/blob/37665ea119c94b2deb864e561dd68f6c016887e5/pkg/apis/garden/validation/validation.go#L1133

I'll suggest to remove the regular expression and use that function.

@rfranzke Any reason why we use this regex or it's just a leftover?

Reply:
Agreed, probably a left over. Let's use the DNS1123Subdomain validation function of the API machinery: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L134

@matthew-marchetti could you adapt your PR to use this function and squash your commits, please?

Thank you!